ceph_iscsi_config/lun.py

```diff
@@ -640,6 +640,14 @@ def add_dev_to_lio(self, in_wwn=None):
             new_lun.delete()
             return None
 
+        try:
+            new_lun.set_attribute("max_data_area_mb",
+                                  settings.config.max_data_area_mb)
+        except RTSLibError as err:
+            self.logger.warning("Could not set LIO device attribute "
+                                "max_data_area_mb for device: {} - "
+                                "({})".format(self.config_key, str(err)))
+
         self.logger.info("(LUN.add_dev_to_lio) Successfully added {}"
```

Comment:
This will always fail because max_data_area_mb is not a writable attribute. We need it set before enablement, so it is a control setting. This is why you need the rtslib patch

https://github.com/open-iscsi/rtslib-fb/pull/112/files

and pass it in at object creation time.

Reply:
I have to add support to targetcli so I will fix this too in my other PR.

Reply:
There package updates is not ideal, but understood. It would be nice if rtslib_fb didn't instantly enable the LUN at instantiation so that the caller could tweak it before enabling.